src/coins.cpp

```diff
@@ -95,6 +96,15 @@ void CCoinsViewCache::AddCoin(const COutPoint &outpoint, Coin&& coin, bool possi
     it->second.coin = std::move(coin);
     it->second.flags |= CCoinsCacheEntry::DIRTY | (fresh ? CCoinsCacheEntry::FRESH : 0);
     cachedCoinsUsage += it->second.coin.DynamicMemoryUsage();
+    TRACE8(utxocache, add,
+           outpoint.hash.ToString().c_str(),
```

Comment:
do we need the hash string here (and the ones below)? we should avoid doing non-negligible computations in traces.

Reply:
We need if we want to support the current versions of `bpftrace` as it can't print binary data in hex (or at all).

related discussion: https://github.com/bitcoin/bitcoin/pull/22006#discussion_r638775357

Reply:
I agree that `outpoint.hash.ToString().c_str()` here is called _a lot_ more than compared to the `hash.data().toString().c_str()` in the connect_block tracepoint. This could hurt performance. Might need to drop this.

Reply:
How about only using the first 64bit of the hash: `outpoint.hash.GetUint64(0)`. This should be very fast and maybe unique enough, at least for the purpose of tracing

Reply:
We already pass `outpoint.hash.data()` as the last argument, that's not the issue. The issue here is the `ToString()` part which might be to expensive in such a tight loop and is a work around for `bpftrace` not supporting hex-formatting.

Reply:
I just came across https://github.com/iovisor/bpftrace/issues/659#issuecomment-568624842. We don't need to use the built-in hex-formatting as it's quite simple just `printf("%02x")` each byte of the txid in an [`unroll () {...}`][1]. This even allows us to print the txids in big-endian (the block-explorer format). 

```
  $p = $txid + 31;
  unroll(32) {
      $b = *(uint8*)$p;
      printf("%02x", $b);
      $p-=1;
  }
```
[1]: https://github.com/iovisor/bpftrace/blob/master/docs/reference_guide.md#9-unroll---unroll

This means we can drop the `outpoint.hash.ToString().c_str()` from the tracepoints added here and use the `outpoint.hash.data()` as the first argument.

Reply:
Nice, I was trying to use unroll like this for printing data in bpftrace but could never figure it out. I was never able to do it based on a length argument, probably because it wasn't a literal. This case works fine though since we know the length of the hash.

Reply:
> We already pass outpoint.hash.data() as the last argument, that's not the issue. The issue here is the ToString() part which might be to expensive in such a frequently called function while being only a work around for bpftrace not supporting hex-formatting.

I agree here. Let's remove this ToString argument. It's too much overhead to allocate, construct, deallocate a string here every time even if the tracepoint is disabled.